mergesort.rb

```diff
@@ -1,26 +1,28 @@
 def mergesort(a)
-    # if the array size is 0|1 then the list is considered sorted, return sorted array
-    # split the list into two halves
-    # merge sort each half
-    # combine the sorted halves
-end
+  return a if a.length <= 1
+
+  middle = a.length / 2
+  left = a[0,middle]
+  right = a[middle..-1]
 
-def split_array(a)
-    # find the middle index
-    # split the array in half - take a look at take and drop in the ruby docs, or use ranges
-    # return left and right halves of array as separate arrays - did you know a method can return multiple arrays? look it up
+  left = mergesort(left)
+  right = mergesort(right)
+  combine(left, right)
 end
 
-# precondition: a and b are sorted
 def combine(a, b)
-    # create a results array
-    # counters pointing to the index of the smallest elements in each array
-    # check that we have elements to compare
-        # push the smaller element onto the result array
-    # if there are elements left over in a, move them to result
-    # if there are elements left over in b, move them to result
+  result = []
+  until a.empty? || b.empty?
+    if a.first <= b.first
+      result << a.shift
+    else
+      result << b.shift
+    end
+  end
+  result + a + b
 end
```

Comment:
Thanks so much for looking over this.  I will defintely pay more attention to that in the future!  I really appreciate you feedback!